### PR TITLE
Fixes, UrlShortening and more.

### PR DESCRIPTION
**Pull request by garenyondem**

### CODE REVIEW

File: rupload/Services/Azure/BlobService.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public BlobService(BlobServiceConfig blobServiceConfig, IJsonConfigService<BlobS
             {
                 result = false;
             }
-            return result;           
+            return result;
```
Comment (daronyondem) [inline rupload/Services/Azure/BlobService.cs]: Don't tell me you were looking for leading whitespaces :) How the hell did you find these? A plug-in? or just obsessed? :)

File: rupload/Helpers/Request.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                }
+            }
+        }
+
+        public static async Task<T> ShortenUrl<T>(object bodyObject)
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.Timeout = TimeSpan.FromMinutes(5);
+                httpClient.BaseAddress = new Uri("http://ouo.press");
```
Comment (daronyondem) [inline rupload/Helpers/Request.cs]: This class looks like a generic class but includes hardcoded Uri. Either change the class name to specify this is not a generic implementation of an HttpRequest but just for Ouo.Press or make the Uri variable. 

My suggestion is to make this a shortening service and wrap it with an interface to inject in as one of the implementations of the shorteningservice interface. The implementation class can be simple named OuoPressClient or something like that.

File: rupload/Helpers/Request.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+namespace rupload.Helpers
+{
+    public class Request
+    {
+        public static async Task<T> PostAsync<T>(string url, string serializedBody)
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.Timeout = TimeSpan.FromMinutes(5);
+                using (HttpResponseMessage response = await httpClient.PostAsync(url, new StringContent(serializedBody)))
```
Comment (daronyondem) [inline rupload/Helpers/Request.cs]: You don't need to wrap your HttpClient with using statements. See more here; http://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

File: rupload/Model/UrlShorten.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            set
+            {
+                _slug = value;
+            }
+        }
+    }
+
+    public class UrlShortenRequest
+    {
+        public string _token { get; set; } = "Vx4NPAjwR6eK3qmTZ2YSWohfda0ThZ2STNLcPf7C"; // default token may be deprecated in the future
```
Comment (daronyondem) [inline rupload/Model/UrlShorten.cs]: Is this a service access token? If so, it needs to be in a config file.

File: rupload/Model/UrlShorten.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿
+namespace rupload.Services.Model
+{
+    // Whole URL shorten mechanism works only with http://ouo.press/
```
Comment (daronyondem) [inline rupload/Model/UrlShorten.cs]: Understand. We need to make this as an injected shortening service.

File: rupload/Model/UrlShorten.cs (hunk 4 above)
Comment (daronyondem) [inline rupload/Model/UrlShorten.cs]: If this is again a service access token this should not be in the data model. This is something the service requester should handle.

File: rupload/Helpers/Request.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                {
+                    string responseString = await response.Content.ReadAsStringAsync();
+                    return JsonConvert.DeserializeObject<T>(responseString);
+                }
+            }
+        }
+
+        public static async Task<T> ShortenUrl<T>(object bodyObject)
```
Comment (daronyondem) [inline rupload/Helpers/Request.cs]: Why do we have a generics implementation? I see that this methods return type is pretty much strict to a single model as long as this is the implementation of a single service which it looks like it should be.

**Comment by garenyondem** [comment]
Image of Yaktocat

**Pull request merged**